numpy/typing/_scalars.py

```diff
@@ -0,0 +1,37 @@
+from typing import Union, Tuple, Any
+from datetime import datetime, timedelta
+
+import numpy as np
+
+_DatetimeLike = Union[datetime, np.datetime64]
+_TimedeltaLike = Union[timedelta, np.timedelta64]
+
+# NOTE: mypy has special rules which ensures that `int` is treated as
+# a `float` (and `complex`) superclass, however these rules do not apply
+# to its `np.generic` counterparts.
+# Solution: manually add them them to the unions below
+_IntLike = Union[int, np.integer]
+_FloatLike = Union[float, np.floating, np.integer]
```

Comment:
```suggestion
_FloatLike = Union[float, np.floating, _IntLike]
```
And same for the line below